lib/datasource/deb/index.ts

```diff
@@ -0,0 +1,306 @@
+    const hash = createHash('sha256');
+    hash.update(basePackageUrl);
+    const hashedPackageUrl = hash.copy().digest('hex');
+    const downloadDirectory = await fs.ensureCacheDir(
+      cfg.deb.downloadDirectory
+    );
+
+    const extractionDirectory = await fs.ensureCacheDir(
+      cfg.deb.extractionDirectory
+    );
+    const extractedFile = extractionDirectory + '/' + hashedPackageUrl + '.txt';
+    let extractedFileExists = false;
+    try {
+      await fs.access(extractedFile, fs.constants.R_OK);
+      extractedFileExists = true;
+    } catch (e) {
+      //ignore
+    }
+
+    for (let i = 0; i < DebDatasource.compressions.length; i++) {
+      const compression = DebDatasource.compressions[i];
+      const compressedFile =
+        downloadDirectory + '/' + hashedPackageUrl + '.' + compression;
+      try {
+        const wasUpdated = await this.downloadPackageFile(
+          basePackageUrl,
+          compression,
+          compressedFile
+        );
+        if (wasUpdated || !extractedFileExists) {
+          await this.extract(compressedFile, compression, extractedFile);
+        }
+        return extractedFile;
+      } catch (e) {
+        logger.warn(
+          "Couldn't download package file with compression " +
+            compression +
+            ' from ' +
+            basePackageUrl
+        );
+      }
+    }
+
+    throw 'No compression standard worked for ' + basePackageUrl;
+  }
+
+  /**
+   *
+   * @param cfg
+   * @param basePackageUrl
+   * @param compression
+   * @returns a boolean indicating if the file was modified
+   */
+  async downloadPackageFile(
+    basePackageUrl: string,
+    compression: string,
+    compressedFile: string
+  ): Promise<boolean> {
+    let lastTimestamp: Date = null;
+    try {
+      const stats = await fs.stat(compressedFile);
+      lastTimestamp = stats.mtime;
+    } catch (e) {
+      // ignore if the file doesnt exist
+    }
+
+    const packageUri = basePackageUrl + '/Packages.' + compression;
+    logger.debug(
+      'Downloading package file from ' + packageUri + ' as ' + compressedFile
+    );
+    let needsToDownload = true;
+    // we can use If-Modified-Since to avoid that we are redownloaded the file
+    // More information can be found here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since
+    if (lastTimestamp !== null) {
+      const downloadOptions: HttpOptions = {};
+      downloadOptions.headers = {
+        'If-Modified-Since': lastTimestamp.toUTCString(),
+      };
+      const headResult = await this.http.head(packageUri, downloadOptions);
+      needsToDownload = headResult.statusCode !== 304;
+    }
+
+    if (needsToDownload) {
+      const readStream = this.http.stream(packageUri);
+      const writeStream = fs.createWriteStream(compressedFile);
+
+      await fs.pipeline(readStream, writeStream);
+    }
+
+    return needsToDownload;
+  }
+
+  async probeExtractedPackage(
```

Comment:
This is a goot candidate to add cache decorator, sample:

https://github.com/renovatebot/renovate/blob/4f65b57225a42f9deb010f8dbfeb06eb94c0dd69/lib/manager/terraform/lockfile/hash.ts#L61-L69

Reply:
Mhm but what if the server returns a new compressed file, can I manually invalidate the cache entry then?